### ISSUE TITLE
collector timeout missing

### ISSUE THREAD

**Issue by KlavsKlavsen** [BrightcoveOS/Diamond#697]
I've found that sometimes elasticsearch and/or redis collector can hang.. often leaving me with no statistics at all until it gives up (many minutes)..

diamond should have a general "collector timeout" setting IMHO.. instead of each collector having to implement their own timeout handling. I haven't found such.. but perhaps it's there and I just haven't set it correctly? 

also.. I'd like to have diamond collect in parallel.. to speed up collection -and not delay the other collectors because one is hanging f.ex.

**Comment by kormoc** [comment]
This repo has moved to https://github.com/python-diamond/Diamond

Please open a new ticket there if this issue is not resolved with the current code there.

Thanks!
